### PR TITLE
mjtest crate: procedural macros for re-using mjtest cases as cargo tests

### PR DESCRIPTION
**Pull request by problame**

### CODE REVIEW

**Comment by reiner-dolp** [comment]
There are also some clippy warnings that need fixing

**Comment by problame** [comment]
`cargo fmt` has no effect on my machine, though

**Comment by reiner-dolp** [comment]
bors retry

**Comment by reiner-dolp** [comment]
We also have a collition for the mj-test CI jobs

```
fatal: destination path 'mjtest' already exists and is not an empty directory.
```

File: mjtest-rs/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+name = "mjtest"
+version = "0.0.1"
+authors = [
+    "flip1995 <hello@philkrones.com>",
+    "hediet <>",
+    "joshuabach <>",
+    "problame <>",
+    "reiner-dolp <>"
+]
+# edition = "2018" https://users.rust-lang.org/t/how-to-use-proc-macro-on-rust-2018/20833/3
```
Comment (reiner-dolp) [inline mjtest-rs/Cargo.toml]: Is this garbage or intentional?

**Comment by reiner-dolp** [comment]
I am not sure, but maybe we have to add a line to `build` that clones the submodules? The CI does it correctly, but I am not sure about the test infrastructure of the IPD.

```
git submodule update --init --recursive
```

File: mjtest-rs/Cargo.toml (hunk 1 above)
Comment (problame) [reply mjtest-rs/Cargo.toml]: intentional, if someone wonders why we can't use 2018 yet

**Comment by problame** [comment]
>     * I am not sure, but maybe we have to add a line to `build` that clones the submodules? The CI does it correctly, but I am not sure about the test infrastructure of the IPD.
>       ```
>       git submodule update --init --recursive
>       ```

`./build` does not invoke `cargo test`, so nothing fails in speedcenter

**Comment by reiner-dolp** [comment]
Ah, right :) but then we cannot use the build script locally for setup during devevlopment. In this case I just have to remember.

**Comment by reiner-dolp** [comment]
The CI fails because you cannot pass `--release` to `cargo fmt`

**Comment by problame** [comment]
> Ah, right :) but then we cannot use the build script locally for setup during devevlopment. In this case I just have to remember.

I have a re-write of the README.md in https://github.com/comprakt/comprakt/pull/28/commits/fcf46bdda5aa798d492aa6af15f11f07fb6ae37e which will cover `git submdoule update --init --recursive`.

File: .travis.yml [hunk 2]
```diff
@@ -12,48 +12,73 @@ branches:
     # This is enabled to build pull requests.
     - master
 
+env:
+  - TEST_KIND="mjtest-lexer"
```
Comment (flip1995) [inline .travis.yml]: Why do we need this? Everywhere in the matrix the `TEST_KIND` is defined. This shouldn't be required.

File: .travis.yml [hunk 3]
```diff
@@ -12,48 +14,73 @@ branches:
...
-      MJ_TIMEOUT=60 MJ_RUN=target/release/comprakt python3 ./mjtest/mjt.py $MJ_TESTS
-    fi
+  - > 
+    case "$TEST_KIND" in
+      comprakt-*)
+        case "$TEST_KIND" in
+          comprakt-release) FLAGS="--release"
+        esac
+        cargo fmt --all -- --check && \
+        cargo clippy $FLAGS --all-targets --all-features -- -D warnings && \
```
Comment (flip1995) [inline .travis.yml]: We also don't need a `--release` flag on Clippy.

**Comment by reiner-dolp** [comment]
looks good to me now. Great work!

**Pull request merged**